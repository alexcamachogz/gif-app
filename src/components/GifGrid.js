import React from 'react'
import { useFetchGif } from '../hooks/useFetchGif'
import { GifGridItem } from './GifGridItem'
// import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGif(category)
  return (
    <>
      <h3>{category}</h3>
      {loading && <p>Loading...</p>}
      <div className="card-grid">
        {
          images.map(img => (
            <GifGridItem key={img.id} {...img}/>
          ))
        }
      </div>
    </>
  )
}