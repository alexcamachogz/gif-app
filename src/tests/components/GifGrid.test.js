import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGif } from '../../hooks/useFetchGif'

jest.mock('../../hooks/useFetchGif')

describe('Testing <GifGrid />', () => {
  const category = 'One Piece'

  test('Should be show <GifGrid /> properly', () => {
    useFetchGif.mockReturnValue({
      data: [],
      loading: true,
    })
    const wrapper = shallow(<GifGrid category={category}/>)
    expect(wrapper).toMatchSnapshot()
  })

  test('Should be show items when images will be loaded', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'http://localhost/image.jpg',
        title: 'Something',
      }]

    useFetchGif.mockReturnValue({
      data: gifs,
      loading: false,
    })

    const wrapper = shallow(<GifGrid category={category}/>)
    expect(wrapper.find('p').exists()).toBe(false)
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
  })

})