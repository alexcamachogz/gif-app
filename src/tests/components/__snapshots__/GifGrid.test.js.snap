// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifGrid /> Should be show <GifGrid /> properly 1`] = `
<Fragment>
  <h3>
    One Piece
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
